### PR TITLE
add github action to test on windows

### PR DESCRIPTION
**Pull request by yagebu**

### CODE REVIEW

**Comment by yagebu** [comment]
Ref #523 

The Github actions aren't triggered here, but are triggered in the PR on my repo: https://github.com/yagebu/beancount/pull/1

File: beancount/scripts/deps.py [hunk 1]
```diff
@@ -125,7 +125,7 @@ def check_python_magic():
             # 'filemagic' is installed; install python-magic.
             raise ImportError
         return ('python-magic', 'OK', True)
-    except ImportError:
+    except (ImportError, OSError):
         return ('python-magic', None, False)
```
Comment (dnicolodi) [inline beancount/scripts/deps.py]: Why does an import result in an OSError? Can you please submit this patch as a standalone PR?

**Comment by dnicolodi** [comment]
If I understand correctly, you dropped Python 3.5 because it is not supported by Github actions. Is this the case? Can you please squash the two remaining commits?

**Comment by yagebu** [comment]
This should be considered WIP, sorry

**Comment by dnicolodi** [comment]
One nice thing of AppVeyor is that it can be used to produce wheels and these can be easily downloaded and uploaded on PyPI at least it seems, I haven't tried. This could be automated and done at each release. Can Github actions be used easily to achieve the same?

File: beancount/scripts/deps.py (hunk 1 above)
Comment (yagebu) [reply beancount/scripts/deps.py]: Exactly the error in #523

Comment (dnicolodi) [reply beancount/scripts/deps.py]: See my comments in #523

**Comment by yagebu** [comment]
> Can Github actions be used easily to achieve the same?

I believe all the CI solutions are mostly equivalent in their features, so that should also be possible. With Fava, I've also been a mostly happy user of AppVeyor.

Somehow, the builds aren't triggered any more (on my fork), so I'll close this PR. Maybe I'll have another go at this at some point.

**Comment by dnicolodi** [comment]
I am trying to get the tests going on AppVeyoir but so far they get stock somewhere in setup_test.py and I get a lot of tests failures.
